src/hotspot/share/prims/whitebox.cpp

```diff
@@ -2284,21 +2284,23 @@ WB_ENTRY(void, WB_CheckThreadObjOfTerminatingThread(JNIEnv* env, jobject wb, job
   }
 WB_END
 
-WB_ENTRY(void, WB_VerifyFrames(JNIEnv* env, jobject wb, jboolean log))
+WB_ENTRY(void, WB_WalkFrames(JNIEnv* env, jobject wb, jboolean log, jboolean verify_oops))
   intx tty_token = -1;
   if (log) {
     tty_token = ttyLocker::hold_tty();
-    tty->print_cr("[WhiteBox::VerifyFrames] Walking Frames");
+    tty->print_cr("[WhiteBox::WalkFrames] Walking Frames");
   }
-  for (StackFrameStream fst(JavaThread::current(), true, true); !fst.is_done(); fst.next()) {
+  for (StackFrameStream fst(JavaThread::current(), verify_oops, true); !fst.is_done(); fst.next()) {
```

Comment:
Can you elaborate, please, why `verify_oops` is equivalent to `RegisterMap::_update_map`?

Reply:
If the register map is not updated, we can't do the call to `frame::verify` below that, which currently only verifies oops in the frame. I thought that the name `verify_oops` would best describe the functionality of the parameter in general, but maybe I've got that backwards... I use it specifically to turn off `RegisterMap::_update_map` after all.

Reply:
Renaming this to `update_reg_map_and_verify_oops` a bit of a mouth full maybe, but want to make clear that oop verification only happens when the flag is set.